test/gtest/uct/test_md.cc

```diff
@@ -937,33 +937,49 @@ UCS_TEST_SKIP_COND_P(test_md, rkey_compare, !check_caps(UCT_MD_FLAG_REG))
     size_t size                      = 4096;
     void *address                    = NULL;
     uct_rkey_compare_params_t params = {};
-    std::vector<uint8_t> rkey_buffer(md_attr().rkey_packed_size + 1);
+    std::vector<uint8_t> rkey_buffer1(md_attr().rkey_packed_size + 1);
+    std::vector<uint8_t> rkey_buffer2(md_attr().rkey_packed_size + 1);
     uct_rkey_bundle_t b1, b2;
     ucs_status_t status;
-    uct_mem_h memh;
-    int result, expect_result;
+    uct_mem_h memh1, memh2;
+    int result1, result2;
+
+    if (GetParam().md_name == "cuda_ipc") {
+        UCS_TEST_SKIP_R("missing IPC capability for registration");
+    }
 
     status = ucs_mmap_alloc(&size, &address, 0, "test_rkey_compare_equal");
     ASSERT_UCS_OK(status);
-    status = reg_mem(UCT_MD_MEM_ACCESS_ALL, address, size, &memh);
+    status = reg_mem(UCT_MD_MEM_ACCESS_ALL, address, size, &memh1);
+    ASSERT_UCS_OK(status);
+    status = reg_mem(UCT_MD_MEM_ACCESS_ALL, address, size, &memh2);
     ASSERT_UCS_OK(status);
-    status = uct_md_mkey_pack(md(), memh, &rkey_buffer[0]);
+    status = uct_md_mkey_pack(md(), memh1, &rkey_buffer1[0]);
+    ASSERT_UCS_OK(status);
+    status = uct_md_mkey_pack(md(), memh2, &rkey_buffer2[0]);
     ASSERT_UCS_OK(status);
 
-    status = uct_rkey_unpack(GetParam().component, &rkey_buffer[0], &b1);
+    status = uct_rkey_unpack(GetParam().component, &rkey_buffer1[0], &b1);
     ASSERT_UCS_OK(status);
-    status = uct_rkey_unpack(GetParam().component, &rkey_buffer[0], &b2);
+    status = uct_rkey_unpack(GetParam().component, &rkey_buffer2[0], &b2);
     ASSERT_UCS_OK(status);
 
     status = uct_rkey_compare(GetParam().component, b1.rkey, b2.rkey, &params,
-                              &result);
+                              &result1);
+    EXPECT_UCS_OK(status);
+    status = uct_rkey_compare(GetParam().component, b2.rkey, b1.rkey, &params,
+                              &result2);
     EXPECT_UCS_OK(status);
-    expect_result = b1.rkey > b2.rkey ? 1 : b1.rkey < b2.rkey ? -1 : 0;
-    EXPECT_EQ(expect_result, result);
+
+    EXPECT_TRUE(((result1 > 0) && (result2 < 0)) ||
```

Comment:
maybe also check that comparing rkey to itself returns 0?

Reply:
added one self comparison